src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -154,6 +155,12 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
                     descriptors.Add(new ParameterDescriptor(ScriptConstants.SystemExecutionContextParameterName, typeof(ExecutionContext)));
                 }
 
+                if (!descriptors.Any(p => p.Type == typeof(CancellationToken)))
+                {
+                    // Add the CancellationToken. We need this to ensure we can cancel even if the Function hasn't requested it.
```

Comment:
We won't have access to it for watching/canceling. It needs to be part of the descriptor to be in the parameter array that gets passed to Invoke.

---
In reply to: [75969178](https://github.com/Azure/azure-webjobs-sdk-script/pull/604#discussion_r75969178) [](ancestors = 75969178)